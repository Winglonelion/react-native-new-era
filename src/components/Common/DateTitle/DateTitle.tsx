import React, { memo } from 'react';
import { StyleSheet, View } from 'react-native';
import { format } from 'date-fns';
import TitleText, {
  TitleTextPropTypes,
} from 'components/Base/Text/TitleText/TitleText';

interface PropTypes extends TitleTextPropTypes {
  date?: Date;
}

const DateTitle: React.FC<PropTypes> = memo(
  ({ level = 'h3', date = new Date() }) => {
    return (
      <View>
        <TitleText level={level} style={styles.text}>
          {format(date, 'eee dd, yyyy')}
        </TitleText>
      </View>
    );
  },
);

const styles = StyleSheet.create({
  text: { textAlign: 'center' },
});

export default DateTitle;
