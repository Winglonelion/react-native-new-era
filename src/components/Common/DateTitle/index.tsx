export { default as default } from './DateTitle';
