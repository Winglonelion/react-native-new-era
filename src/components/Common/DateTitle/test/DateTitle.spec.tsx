import React from 'react';
import { shallow } from 'enzyme';
import DateTitle, { PropTypes } from '../DateTitle';

const setup = (propOverrides: PropTypes = {} as PropTypes) => {
  const props = { ...propOverrides };
  const wrapper = shallow(<DateTitle {...props} />);
  return {
    props,
    wrapper,
  };
};

describe('DateTitle check snapshot', () => {
  test('should match snapshot with props', () => {
    const { wrapper } = setup({
      color: 'red',
      date: new Date(),
      level: 'h2',
      style: { lineHeight: 30 },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('should match snapshot without props and loading', () => {
    const { wrapper } = setup({});
    expect(wrapper).toMatchSnapshot();
  });
});
