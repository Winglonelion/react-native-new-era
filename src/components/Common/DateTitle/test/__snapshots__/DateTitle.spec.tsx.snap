// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTitle check snapshot should match snapshot with props 1`] = `
<View>
  <Memo()
    level="h2"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Tue 07, 2021
  </Memo()>
</View>
`;

exports[`DateTitle check snapshot should match snapshot without props and loading 1`] = `
<View>
  <Memo()
    level="h3"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Tue 07, 2021
  </Memo()>
</View>
`;
